pysatModels/models/pydineof_dineof.py

```diff
@@ -0,0 +1,248 @@
+                               supported_tags=supported_tags)
+
+
+def load(fnames, tag=None, inst_id=None, **kwargs):
+    """Loads pydineof data using xarray.
+
+    This routine is called as needed by pysat. It is not intended
+    for direct user interaction.
+
+    Parameters
+    ----------
+    fnames : array-like
+        iterable of filename strings, full path, to data files to be loaded.
+        This input is nominally provided by pysat itself.
+    tag : string ('')
+        tag name used to identify particular data set to be loaded.
+        This input is nominally provided by pysat itself.
+    inst_id : string ('')
+        Instrument ID used to identify particular data set to be loaded.
+        This input is nominally provided by pysat itself.
+    **kwargs : extra keywords
+        Passthrough for additional keyword arguments specified when
+        instantiating an Instrument object. These additional keywords
+        are passed through to this routine by pysat.
+
+    Returns
+    -------
+    data : xarray.Dataset
+        pysat formatted xarray Dataset
+    meta : pysat.Metadata
+        Model run meta data
+
+    Note
+    ----
+    Any additional keyword arguments passed to pysat.Instrument
+    upon instantiation are passed along to this routine.
+
+    ::
+
+        inst = pysat.Instrument(inst_module=pysatModels.models.pydineof_dineof)
+        inst.load(2019, 1)
+
+    """
+
+    # netCDF4 files were produced by xarray
+    # returning an xarray.Dataset
+    return pysat.utils.load_netcdf4(fnames, epoch_name='time',
+                                    pandas_format=False)
+
+
+def download(date_array=None, tag=None, inst_id=None, data_path=None, **kwargs):
+    """Downloads dineof data.  Currently only retrieves test data from github
+
+    Parameters
+    ----------
+    date_array : array-like
+        list of datetimes to download data for. The sequence of dates need not
+        be contiguous.
+    tag : string
+        Tag identifier used for particular dataset. This input is provided by
+        pysat. (default='')
+    inst_id : string
+        Instrument ID string identifier used for particular dataset. This input
+        is provided by pysat. (default='')
+    data_path : string
+        Path to directory to download data to. (default=None)
+    **kwargs : dict
+        Additional keywords supplied by user when invoking the download
+        routine attached to a pysat.Instrument object are passed to this
+        routine via kwargs.
+
+    Note
+    ----
+    This routine is invoked by pysat and is not intended for direct use by
+    the end user.
+
+    The test object generates the datetime requested by the user, which may not
+    match the date of the model run.
+
+    """
+
+    if tag == 'test':
+        date = date_array[0]
+        remote_url = 'https://github.com/pysat/pysatModels/'
+        remote_path = 'blob/pysatDINEOF_inst/pysatModels/tests/test_data/'
```

Comment:
I had to put the reference for this file in with the current branch to get things to work.